json-schema/schema.json

```diff
@@ -310,14 +299,110 @@
         },
         "mlm:hyperparameters": {
           "$ref": "#/$defs/mlm:hyperparameters"
+        },
+        "mlm:artifact_type": {
+          "$ref": "#/$defs/mlm:artifact_type"
         }
       },
-      "$comment": "Allow properties not defined by MLM prefix to allow combination with other extensions.",
+      "$comment": "Allow properties not defined by MLM prefix to work with other extensions and attributes, but disallow undefined MLM fields.",
       "patternProperties": {
         "^(?!mlm:)": {}
       },
       "additionalProperties": false
     },
+    "mlmCollectionFields": {
+      "description": "Schema to validate the MLM fields permitted under Collection summaries.",
+      "allOf": [
+        {
+          "description": "Fields that are mandatory under the Collection summaries.",
+          "type": "object",
+          "required": []
+        },
+        {
+          "description": "Fields that are disallowed under the Collection summaries.",
+          "not": {
+            "required": [
+              "mlm:input",
```

Comment:
should other fields that are specific to one model be disallowed under collections? like `mlm:total_parameters` or `mlm:memory_size`

Reply:
Yes, there could definitely be more. Could be done in another PR. Easily extensible now with how the schema was reworked.

`mlm:memory_size`, I would leave available for override in the assets since different frameworks could result in different results. I'm not sure about `mlm:total_parameters`. Makes sense when referring to the generic number (eg. 2B), but could this number vary slightly when using different frameworks or compile/optimization?

Reply:
off the top I think the total_parameter size could change if the optimization was a full retraining with a student/teacher set up to compress a larger teacher model to a smaller model. I think leaving these available to override in assets makes sense. Not sure if they will ever/ should ever be used in Collection level metadata. I'm fine for this to wait for another PR